sdks/java/transform-service/src/main/java/org/apache/beam/sdk/transformservice/ExpansionService.java

```diff
@@ -48,10 +61,61 @@ public class ExpansionService extends ExpansionServiceGrpc.ExpansionServiceImplB
         clientFactory != null ? clientFactory : DEFAULT_EXPANSION_SERVICE_CLIENT_FACTORY;
   }
 
+  // Waits till all expansion services are ready.
+  private void waitForAllServicesToBeReady() throws TimeoutException {
+    if (disableServiceCheck) {
+      // Service check disabled. Just returning.
+      return;
+    }
+
+    outer:
+    for (Endpoints.ApiServiceDescriptor endpoint : endpoints) {
+      long start = System.currentTimeMillis();
+      long duration = 10;
+      while (System.currentTimeMillis() - start < SERVICE_CHECK_TIMEOUT) {
+        try {
+          String url = endpoint.getUrl();
+          int portIndex = url.lastIndexOf(":");
+          int port = portIndex > 0 ? Integer.parseInt(url.substring(portIndex + 1)) : 80;
```

Comment:
Is default port 80 documented somewhere / rely on some other components ? If I remember correctly the port for expansion service is dynamically signed?

Reply:
Thanks. Updated.